iosapp/app/Lib/Animation.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+import SwiftUI
+
+func delayed(by delay: Duration, _ callback: @escaping () -> Void) {
+  DispatchQueue.main.asyncAfter(deadline: .now() + delay.inMilliseconds / 1000) {
+    callback()
+  }
+}
+
+extension Duration {
+  var inMilliseconds: Double {
+    let v = components
+    return Double(v.seconds) * 1000 + Double(v.attoseconds) * 1e-15
+  }
+}
+
+struct Vector: Equatable, Identifiable, AdditiveArithmetic {
+  var x: Double
+  var y: Double
+
+  var magnitude: Double {
+    sqrt(self.x * self.x + self.y * self.y)
+  }
+
+  var isOrigin: Bool {
+    self.x == 0 && self.y == 0
+  }
+
+  var normalized: Vector {
+    self.scaledBy(1 / self.magnitude)
+  }
+
+  func scaledBy(_ scalar: Double) -> Vector {
+    .init(x: self.x * scalar, y: self.y * scalar)
+  }
+
+  static var zero: Vector {
+    .init(x: 0, y: 0)
+  }
+
+  static func - (lhs: Vector, rhs: Vector) -> Vector {
+    .init(x: lhs.x - rhs.x, y: lhs.y - rhs.y)
+  }
+
+  static func + (lhs: Vector, rhs: Vector) -> Vector {
```

Comment:
very cool. swift shines here with the ease of implementing arithmetic operations on a type like this.  did you know that there's a type coming to the std library fairly soon maybe called Vector? don't think it affects this code at all, but there's tons of bike-shedding about naming and whether it's a mathematical vector, and how rust/c++ name things: https://forums.swift.org/t/vector-a-fixed-size-array/75264